Automattic/pocket-casts-android
# Onboarding Welcome Screen

| 📘 Project: #524  |
| --- |

## Description
Adding the welcome screen to the onboarding flow.

Note that the welcome screen does not include the card for importing podcasts because the import screens have not been completed yet.

## Testing Instructions

For all of these tests, make sure you turn on the onboarding feature flag in `base.gradle, and then complete these steps to load the welcome screen:
1. Open the app when not signed into an account
2. Tap "Sign Up"
3. Create an account
4. Tap "Not now" on the placeholder recommendations screen
5. ✅ Observe that the screens match the designs for both a Plus user and a free account user
6. ✅ Observe that the newsletter toggle is "on" by default

### 1. Discover something new

1. Tap on the "Discover something new" card
2. ✅ Observe that the main app screen is opened **to the Discover tab**
3. ✅ Go to the account screen and verify that the toggle state from the welcome page was persisted

### 2. Done

1. Tap on the "Done" button
2. ✅ Observe that the main app screen is opened
3. ✅ Go to the account screen and verify that the toggle state from the welcome page was persisted

### 3. Swipe back

1. Swipe back
2. ✅ Observe that a Plus user is taken back to the placeholder recommendations screen and a free user is taken back to the upgrade screen
3. Swipe back until you are taken back to the app's main screen
4. ✅ Go to the account screen and verify that the toggle state from the welcome page was persisted

### 4. Newsletter toggle removed from Create Account Page
1. Open the app when not logged in
5. Tap Create Account
7. ✅ Observe that there is not any option to change the newsletter (it moved from this screen to the welcome screen in this PR)

## Screenshots or Screencast 

| Free | Plus |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/4656348/203136190-fb5fd779-e241-4709-a9cc-765888e3f9ae.png) | ![image](https://user-images.githubusercontent.com/4656348/203136135-08dcad0f-44ad-451e-936e-7455bd1e8e75.png) |


## Checklist
 
#### I have tested any UI changes...
